exec-analysis/src/main/java/com/capitalone/dashboard/exec/collector/PortfolioCollector.java

```diff
@@ -359,4 +400,45 @@ public void run() {
     public void onStartup() {
         taskScheduler.schedule(this, new CronTrigger(setting.getCron()));
     }
+
+    public String getPortfolioWithThumbnails(List<Portfolio> portfolioList) {
+        ldapHandler.initLdaptemplate();
+        setLdapTemplate(LdapHandler.getLdaptemplate());
+        PortfolioThumbnail portfolioThumbnail = null;
+        List<PortfolioThumbnail> portfolioListThumbnail = new ArrayList<>();
+        for(Portfolio p :  portfolioList) {
+            portfolioThumbnail = new PortfolioThumbnail();
+            String pName = p.getName();
+            String lob = p.getLob();
+            String photoString = null;
+            String lastName = pName.substring(pName.indexOf(" ") + 1);
+            String firstName = pName.substring(0, pName.indexOf(" "));
+            String fullName = lastName + ", " + firstName;
+            if (!fullName.isEmpty()) {
+                List<Person> execInfo = findByName(fullName);
+                if (execInfo != null && !execInfo.isEmpty()) {
+                    byte[] photo = (byte[]) execInfo.get(0).getThumbnailPhoto();
+                    if (photo == null) {
+                        photoString = "";
+                    } else {
+                        photoString = Base64.getEncoder().encodeToString(photo);
+                    }
+                }
+                LOGGER.info("Byte code convert to string :" + photoString);
+            } else {
+                LOGGER.info("List size is empty");
+            }
+            portfolioThumbnail.setName(pName);
+            portfolioThumbnail.setLob(lob);
+            portfolioThumbnail.setThumbnail(photoString);
+            portfolioListThumbnail.add(portfolioThumbnail);
+        }
+        portfolioRepositoryThumbnail.save(portfolioListThumbnail);
+        return null;
```

Comment:
Should not be returning null.

Reply:
I removed the null after save, there is no return for the function.